src/interactions/slashCommands/SlashCommandBuilder.ts

```diff
@@ -0,0 +1,119 @@
+	 * creating slash commands via this builder.
+	 *
+	 * **Note:** Calling this getter will validate required properties based on their conditions.
+	 */
+	public toJSON() {
+		validateRequiredParameters(this.name, this.description, this.options);
+		return {
+			name: this.name,
+			description: this.description,
+			options: this.options.map((option) => option.toJSON()),
+		};
+	}
+
+	/**
+	 * Adds a new sub command group to this command
+	 * @param input A function that returns a sub command group builder, or an already built builder
+	 */
+	public addSubCommandGroup(
+		input:
+			| SlashCommandSubCommandGroupBuilder
+			| ((subCommandGroup: SlashCommandSubCommandGroupBuilder) => SlashCommandSubCommandGroupBuilder),
+	): SlashCommandSubCommandGroupsOnlyBuilder {
+		const { options } = this;
+
+		// First, assert options conditions - we cannot have more than 25 options
+		validateMaxOptionsLength(options);
+
+		// Make sure there is no sub command at the root level - if there is, throw
+		const hasSubCommands = options.some((item) => item instanceof SlashCommandSubCommandBuilder);
+		if (hasSubCommands) throw new RangeError(`You cannot mix sub commands and sub command groups at the root level.`);
+
+		// Get the final result
+		const result = typeof input === 'function' ? input(new SlashCommandSubCommandGroupBuilder()) : input;
+
+		if (!(result instanceof SlashCommandSubCommandGroupBuilder))
+			throw new TypeError(
+				`Expected to receive a SlashCommandSubCommandGroupBuilder back, got ${
+					// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+					((result as any)?.name as string) ?? 'Unknown'
+				} ("${typeof result}") instead`,
+			);
+
+		// Push it
+		options.push(result);
+
+		return this;
+	}
+
+	/**
+	 * Adds a new sub command to this command
+	 * @param input A function that returns a sub command builder, or an already built builder
+	 */
+	public addSubCommand(
+		input:
+			| SlashCommandSubCommandBuilder
+			| ((subCommandGroup: SlashCommandSubCommandBuilder) => SlashCommandSubCommandBuilder),
+	): SlashCommandSubCommandsOnlyBuilder {
+		const { options } = this;
+
+		// First, assert options conditions - we cannot have more than 25 options
+		validateMaxOptionsLength(options);
+
+		// Make sure there is no sub command at the root level - if there is, throw
+		const hasSubCommandGroups = options.some((item) => item instanceof SlashCommandSubCommandGroupBuilder);
+		if (hasSubCommandGroups)
+			throw new RangeError(`You cannot mix sub commands and sub command groups at the root level.`);
+
+		// Get the final result
+		const result = typeof input === 'function' ? input(new SlashCommandSubCommandBuilder()) : input;
+
+		if (!(result instanceof SlashCommandSubCommandBuilder))
+			throw new TypeError(
+				`Expected to receive a SlashCommandSubCommandBuilder back, got ${
+					// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+					((result as any)?.name as string) ?? 'Unknown'
```

Comment:
Same here about the `.constructor` thing.

Reply:
To be honest, this should be moved into an utility function and use [`@@toStringTag`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) when possible, so we can emulate the type from Node.js's `util.inspect`, which is `${ValueConstructorName}[ \[${ValueToStringTag}\]]`, e.g. `Client` and `Collection [Map]`.

The function would also abstract those casts away, improving the readability of those parts, as well as reducing code duplication.